Source/WPF/MyMoney/Database/Money.cs

```diff
@@ -8254,59 +8254,6 @@ static decimal GetInvestmentValue(System.Collections.IEnumerable data, out decim
             return balance;
         }
 
-        public static decimal GetBalance(MyMoney money, Account a, Category category, System.Collections.IEnumerable data, out decimal salestax, out decimal investmentValue)

```

Comment:
In general I don't like the idea of removing helper methods like this one and moving more logic in the UI layer (TransactionView.xaml.cs).  You'll see that TransactionView.xaml.cs is already too big, so I don't like it getting bigger.  But your new code for GetBalance is dependent on TransactionViewName.  But I don't think it needs to be.  I think you can keep this helper method and add one Boolean flag "CurrencyNormalized".  If it is then do this:
```
balance += t.CurrencyNormalizedAmount(t.AmountMinusTax);
salestax += t.CurrencyNormalizedAmount(t.NetSalesTax);
```
otherwise do the default thing:
```                         
balance += t.Amount;
salestax += t.NetSalesTax;
```
I also think the addition of InvestmentTaxes is fine in all cases, you can have a "ByCategory" transaction also be an Investment transaction, so in that case you are losing the taxes listed in the investment.

Reply:
The helper function was going to end up with 8 parameters so I though it was cleaner to just merge it in.  I'll go back and rework to make it a helper.

Reply:
the result, and all the out parameters could be bundled into a Tuple or a new struct.